### PR TITLE
provisioning_template finds templates by symbol

### LINKED ISSUE #40: setting host to build mode fails with error message
When setting a host to build mode, it currently always fails with this error message:

```
No PXELinux templates were found for this host, make sure you define at least one in your CoreOS 1967.4.0 settings or change PXE loader
```

### PR DESCRIPTION
**Pull request by timogoebel**
Fixes #40.

### CODE REVIEW

File: test/models/concerns/foreman_git_templates/orchestration/tftp_test.rb [hunk 1]
```diff
@@ -62,8 +62,47 @@ class TFTPOrchestrationTest < ActiveSupport::TestCase
...
+          end
+
+          host.provision_interface.send(:validate_tftp)
+          assert_empty host.errors.messages
+        end
+      end
+    end
+
+    context 'host is not build mode' do
+      it 'validates that the host is ready for tftp' do
```
Comment (kamils-iRonin) [inline test/models/concerns/foreman_git_templates/orchestration/tftp_test.rb]: should this test be exactly the same as test for build mode?

File: test/models/concerns/foreman_git_templates/orchestration/tftp_test.rb [hunk 2]
```diff
@@ -62,8 +62,47 @@ class TFTPOrchestrationTest < ActiveSupport::TestCase
     end
   end
 
+  describe '#validate_tftp' do
+    let(:host) { FactoryBot.create(:host, :with_tftp_orchestration, :with_template_url, pxe_loader: 'PXELinux BIOS') }
```
Comment (kamils-iRonin) [inline test/models/concerns/foreman_git_templates/orchestration/tftp_test.rb]: let's move `pxe_loader: 'PXELinux BIOS'` to the trait

File: test/models/concerns/foreman_git_templates/orchestration/tftp_test.rb (hunk 1 above)
Comment (timogoebel) [reply test/models/concerns/foreman_git_templates/orchestration/tftp_test.rb]: Yeah, we want to run it both for build mode and not for build mode. We could do `.each` but I think that doesn't improve readability.

File: test/models/concerns/foreman_git_templates/orchestration/tftp_test.rb (hunk 2 above)
Comment (timogoebel) [reply test/models/concerns/foreman_git_templates/orchestration/tftp_test.rb]: Good idea, but it's actually not related, is it?

**Pull request merged**